demos/shadow/shadow_demo_main/shadow_demo_main.c

```diff
@@ -173,6 +200,24 @@ static uint32_t clientToken = 0U;
  */
 static bool eventCallbackError = false;
 
+/**
+ * @brief Status of the response of Shadow delete operation from AWS IoT
+ * message broker.
+ */
+static bool deleteResponseReceived = false;
+
+/**
+ * @brief Status of the Shadow delete operation.
+ *
+ * The Shadow delete status will be updated by the incoming publishes on the
+ * MQTT topics for delete acknowledgement from AWS IoT message broker
+ * (accepted/rejected). Shadow document is considered to be deleted if an
+ * incoming publish is received on `/delete/accepted` topic or an incoming
+ * publish is received on `/delete/rejected` topic with error code 404. Code 404
+ * indicates that the Shadow document does not exist for the Thing yet.
+ */
+static bool shadowDeleted = false;
```

Comment:
What is the significance of this flag in the execution of the demo? Is there an update to the flow of the demo based on the value of the flag?

Reply:
That is a good catch. Demo will be marked as success or failure based on if a delete is successful. Missed that logic in the demo. Will add that. Thanks